### PR TITLE
Add hystrix implementation at Model & Feast call

### PR DESCRIPTION
**Pull request by ariefrahmansyah**
**What this PR does / why we need it**:

To improve the reliability and scalability of the standard transformer, we would like to integrate Hystrix when calling Model and Feast.

- Add Hystrix when calling Merlin model
- Add Hystrix when fetching Feast features

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

- [x] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: api/pkg/transformer/feast/feature_retriever_test.go [hunk 1]
```diff
@@ -2654,3 +2656,107 @@ func Benchmark_buildEntitiesRequest_geohashArrays(b *testing.B) {
...
+				atomic.AddUint32(&good, 1)
+			}
+		}()
+
+		time.Sleep(10 * time.Millisecond)
+	}
+
+	time.Sleep(2 * time.Second)
+
+	assert.GreaterOrEqual(t, atomic.LoadUint32(&bad), uint32(0))
```
Comment (pradithya) [inline api/pkg/transformer/feast/feature_retriever_test.go:2758]: It seems the check is too lenient, is it possible to just use equal?

File: api/pkg/transformer/server/server.go [hunk 2]
```diff
@@ -13,36 +13,46 @@ import (
...
 	Port            string `envconfig:"MERLIN_TRANSFORMER_PORT" default:"8081"`
 	ModelName       string `envconfig:"MERLIN_TRANSFORMER_MODEL_NAME" default:"model"`
 	ModelPredictURL string `envconfig:"MERLIN_TRANSFORMER_MODEL_PREDICT_URL" default:"localhost:8080"`
 
 	HTTPServerTimeout time.Duration `envconfig:"HTTP_SERVER_TIMEOUT" default:"30s"`
-	HTTPClientTimeout time.Duration `envconfig:"HTTP_CLIENT_TIMEOUT" default:"15s"`
+	HTTPClientTimeout time.Duration `envconfig:"HTTP_CLIENT_TIMEOUT" default:"1s"`
+
+	ModelTimeout                       time.Duration `envconfig:"MODEL_TIMEOUT" default:"1s"`
+	ModelHystrixMaxConcurrentRequests  int           `envconfig:"MODEL_HYSTRIX_MAX_CONCURRENT_REQUESTS" default:"100"`
```
Comment (pradithya) [inline api/pkg/transformer/server/server.go:46]: Just to be safe, I think we should increase `ModelHystrixMaxConcurrentRequests` to 500

File: api/pkg/transformer/feast/feature_retriever.go [hunk 3]
```diff
@@ -206,95 +230,88 @@ func addSeries(entityName string, series []*types.Value, entities []feast.Row) [
...
+		return batchResult{featuresData: nil, err: err}
+	}
+	feastLatency.WithLabelValues("success").Observe(float64(durationMs))
 
-			if fr.options.CacheEnabled {
-				if err := insertMultipleFeaturesToCache(fr.cache, entityFeaturePairs, project, fr.options.CacheTTL); err != nil {
-					fr.logger.Error("insert_to_cache", zap.Any("error", err))
-				}
-			}
+	fc.logger.Debug("feast_response", zap.Any("feast_response", feastResponse.Rows()))
```
Comment (pradithya) [inline api/pkg/transformer/feast/feature_retriever.go:293]: Can this be removed? even though it's debug `feastResponse.Rows()` can still be executed.

**Comment by pradithya** [comment]
Other than comments above the rest looks good to me.

File: api/pkg/transformer/feast/feature_retriever_test.go (hunk 1 above)
Comment (ariefrahmansyah) [reply api/pkg/transformer/feast/feature_retriever_test.go]: Updated the implementation to be more consistent. Previously, we use the error channel from Hystrix's GoC function. Now we use fallback function that sends the error to batchResultChan. Please refer to: https://github.com/gojek/merlin/blob/transformer-hystrix/api/pkg/transformer/feast/feature_retriever.go#L417-L420

**Review by pradithya — APPROVED** [review]
LGTM!

**Pull request merged**